Automattic/woocommerce-payments
# Update Dockerfile image to `wordpress:php7.4`

Fixes #4942

#### Changes proposed in this Pull Request



**Update Dockerfile image to `wordpress:php7.4`**



This PR updates the image that Docker uses to build the WCPay client environment containers from `wordpress:php7.3` to `wordpress:php7.4`. This change is needed because PHP 7.3 is past its end of life. This PR changes the PHP version that runs in the WCPay client environment containers from PHP 7.3 to PHP 7.4. 

The other solution that was considered was to change the image to `wordpress:php8.0`, but after discussing it with @luizreis, we decided it would be best to just bump the version to 7.4 and start a discussion with the wider team about the eventual update to PHP 8.0.



This code shouldn't break as it is just changing which Docker image is pulled during the Docker build process. That said, it would be possible for WCPay or dependencies to break if they weren't already tested to run on PHP 7.4. To make sure this doesn't break, the build process was tested, all manual testing passed, all JS tests passed, and all PHP tests passed.



#### Testing instructions





* This change doesn't impact end-user functionality. 
* Check out this branch.
* Bring down your client's docker containers if they are running: `npm run down`.
* Bring the containers up: `npm run up:recreate`.
* Build all the things: `npm run build && npm run build:client && npm run postinstall`
* Run JS and PHP tests: `npm run test`. All tests should pass.
* Visit your local client site and login: `http://localhost:8082/wp-login.php`
* Check WCPay admin pages, check that you can view individual transactions, etc.
* Check WooCommerce status page to make sure there are no issues reported.
* Place a test order to ensure it processes successfully.

-------------------


**Post merge**
